### ISSUE TITLE
[Bug] Upon case merge: missing webhook events about the operation performed on merged cases

### ISSUE THREAD

**Issue by martinr103** [TheHive-Project/TheHive#1969]
### Request Type
Bug

### Work Environment

| Question                   | Answer                                |
| -------------------------- | ------------------------------------- |
| OS version (server)        | Debian  |
| OS version (client)        | 10.3           |
| TheHive version / git hash | 4.1.3         |
| Package Type               | Binary / in Docker |
| Browser type & version     | If applicable                         |

### Problem Description
When two cases are merged in Thehive *3* , then the application generates webhook events for:
- creation of the New case (with merged content from the two underlying cases)
- resolution status update for each of the two cases that were merged into a new one -> this webhook event contains a field 'details.mergeInto'

But when two cases are merged in Thehive *4* , then the application generates only an event for the creation of a new case.
There are no webhook events about status change of the two underlying cases.
No information about the resolution / closure / deletion. No webhook event with a "details.mergeInto" field.

These webhook events are essential, if you want to integrate Thehive with another system.

### Steps to Reproduce
1. configure some webhook listener/logger (so that you can see the events)
1. merge two cases in Thehive4
1. check which webhook events have been sent

### Possible Solutions
(keep this section if you have suggestions on how to solve the problem. **Otherwise delete it**)

### Complementary information
(add anything that can help identifying the problem such as **log** excerpts, **screenshots**, **configuration dumps** etc.)

**Comment by martinr103** [comment]
Hello,

Are there any chances, that this could be fixed in 4.1.6 ?

Is it a difficult code change, or just something that was missed out by accident, and easy to re-add ?

(Again, the request is to have webhook events generated for status changes of Cases that were merged INTO some new case)

**Comment by martinr103** [comment]
Hey Guys, are you ignoring this issue report ?
This is probably the oldest, webhook-related bug report for TH4 - opened mid of April, now it is 5 months later.
No comments? Not even an answer if this is an easy or difficult to fix ?

**Comment by martinr103** [comment]
I've done some testing... and can confirm that the webhook event is generated.

Minor thing observed is, that the field "operation" is _not_ "merge" like in your example above.
Instead it is "operation":"update"
Which is a little illogical, as it is the very first event found for the newly created Case (objectId).
I can still use the webhook data (this is the main point) , by ignoring the value in field "operation" and _relying_ on the fact, that the field "details" contains the sub-field called "cases".
This will work well, of course as long as you never produce OTHER webhook events (non-merge) which would contain "**details.cases**".
It would be slightly cleaner and unambiguous, if the field operation was indeed provided as "merge"... but ok... main thing is the webhook is in place at all. Thanks!!

**Comment by nadouani** [comment]
Hello @martinr103 can you share your endpoint declaration (on the config file)

The merge event is available only if the endpoint version is equal to 1. This produces results using v1 format. I can share an example for this.

**Comment by martinr103** [comment]
Oh, Sorry!  I missed the note about "version: 1".  My bad.

Actually, are there any other differences in webhook events (besides this merge stuff), when I switch the "version" from 0 to 1 ??

**Comment by nadouani** [comment]
Yes there are differences. Version 0 is returns backward compatible objects (with TheHive 3) and version 1 contains the object structures of TheHive 4.

**Comment by martinr103** [comment]
Ok, thanks.
Last question to this topic:
The docs page (https://docs.thehive-project.org/thehive/installation-and-configuration/configuration/webhooks/) says:
"Currently TheHive only supports version 0."
Can I safely start using version 1 ?   (being on TH v4.1.12)
